packages/core/src/utils/serializeNode.tsx

```diff
@@ -31,7 +31,7 @@ export const serializeComp = (
         }
         return serializeComp(child, resolver);
       });
-    } else if (prop.type) {
+    } else if (prop.type && typeof prop.props === 'object') {
```

Comment:
The original intention here was to check if the current `prop` is a function and if so we will assume it's a User Component and attempt to serialise it as such. Perhaps, we could just do the following:

```suggestion
    } else if (typeof prop === 'function') {
```

Reply:
I have been running into this too, thanks for working on a fix! Wouldn't it be `typeof prop.type === 'function'` since the .type property has the component/function?